consumer/communication.go

```diff
@@ -45,12 +45,19 @@ func AmfStatusChangeSubscribe(amfUri string, guamiList []models.Guami) (
 	} else if httpResp != nil {
 		if httpResp.Status != localErr.Error() {
 			err = localErr
-			return
+			return nil, err
 		}
 		problem := localErr.(openapi.GenericOpenAPIError).Model().(models.ProblemDetails)
 		problemDetails = &problem
 	} else {
 		err = openapi.ReportError("%s: server no response", amfUri)
 	}
+
+	defer func() {
```

Comment:
What benefit we get with this coding style?

Reply:
why not just close it .. does it give performance benefit of creating new goroutine?

Reply:
Do you mean something like this?
```go
defer httpResp.Body.Close()
```
If so, golangci-linter complains with the following error when just using `defer httpResp.Body.Close()`
```
consumer/communication.go:56:27: Error return value of `httpResp.Body.Close` is not checked (errcheck)
        defer httpResp.Body.Close()
                                 ^
```